### PR TITLE
Full merge of rst support for --list and --avail options

### PR DESCRIPTION
**Pull request by Caylo**
Full merge of PRs #1332 to #1338 

Needs ~~https://github.com/hpcugent/vsc-base/pull/180~~

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/tools/utilities.py [hunk 1]
```diff
@@ -117,3 +120,39 @@ def import_available_modules(namespace):
                     raise EasyBuildError("import_available_modules: Failed to import %s: %s", modpath, err)
                 modules.append(mod)
     return modules
+
+def det_col_width(entries, title):
+    """Determine column width based on column title and list of entries."""
+    return max(map(len, entries + [title]))
+
+def mk_rst_table(titles, values):
```
Comment (boegel) [inline easybuild/tools/utilities.py]: I would keep this in `docs.py`?

File: easybuild/framework/easyconfig/templates.py [hunk 2]
```diff
@@ -36,10 +36,15 @@
...
 from easybuild.tools.build_log import EasyBuildError
 from easybuild.tools.systemtools import get_shared_lib_ext
+from easybuild.tools.utilities import mk_rst_table
 
 
 _log = fancylogger.getLogger('easyconfig.templates', fname=False)
 
+
+FORMAT_RST = 'rst'
+FORMAT_TXT = 'txt'
```
Comment (Caylo) [inline easybuild/framework/easyconfig/templates.py]: :(

File: easybuild/tools/utilities.py (hunk 1 above)
Comment (boegel) [inline easybuild/tools/utilities.py]: to be removed, see https://github.com/hpcugent/vsc-base/pull/180

**Comment by boegel** [comment]
Good to go (finally), thanks @Caylo!

**Pull request merged**